InformationScripting/src/queries/AstQuery.cpp

```diff
@@ -53,14 +53,18 @@ const QStringList AstQuery::NAME_ARGUMENT_NAMES{"n", "name"};
 const QStringList AstQuery::ADD_AS_NAMES{"a", "addAs"};
 const QStringList AstQuery::ATTRIBUTE_NAME_NAMES{"at", "attribute"};
 
-AstQuery::AstQuery(ExecuteFunction exec, Model::Node* target, QStringList args)
-	: ScopedArgumentQuery{target, {
+AstQuery::AstQuery(ExecuteFunction exec, Model::Node* target, QStringList args,
+						 std::vector<Arguments::ArgumentRule> argumentRules)
+	: target_{target}, arguments_{new ArgumentParser({
```

Comment:
You don't need `ArgumentParser` to be a pointer type at all. This is only needed for polymorphism. Use a plain-old type.